src/python/cli_new/lib/cli/mesos.py

```diff
@@ -114,25 +114,25 @@ def get_container_id(task):
         " Please try again.")
 
 
-def get_tasks(master):
+def get_tasks(master, query=None):
     """
     Get the tasks in a Mesos cluster.
     """
     endpoint = "tasks"
     key = "tasks"
 
     try:
-        data = http.get_json(master, endpoint)
+        data = http.get_json(master, endpoint, query)
     except Exception as exception:
         raise CLIException(
-            "Could not open '/{endpoint}' on master: {error}"
-            .format(endpoint=endpoint, error=exception))
+            "Could not open '/{endpoint}' with query: {query} on master: {error}"
+            .format(endpoint=endpoint, query=query, error=exception))
 
     if not key in data:
         raise CLIException(
             "Missing '{key}' key in data retrieved"
-            " from master on '/{endpoint}'"
-            .format(key=key, endpoint=endpoint))
+            " from master on '/{endpoint}' with query: {query}}"
```

Comment:
Same comment as the previous one, returning an exception with `query: None` could be confusing.